### PR TITLE
Misc cleanup

### PR DESCRIPTION
**Pull request by sdlaver**

### CODE REVIEW

**Comment by sdlaver** [comment]
Will be rebased on top of #360 once that is merged.

File: seedvault/src/main/java/com/solanamobile/seedvault/Bip32DerivationPath.java [hunk 1]
```diff
@@ -123,7 +123,7 @@ public static Bip32DerivationPath fromUri(@NonNull Uri bip32Uri) {
             throw new UnsupportedOperationException("BIP32 URI must be hierarchical");
         }
 
-        if (bip32Uri.isAbsolute() && !bip32Uri.getScheme().equals(WalletContractV1.BIP32_URI_SCHEME)) {
```
Comment (sdlaver) [inline seedvault/src/main/java/com/solanamobile/seedvault/Bip32DerivationPath.java:126]: Can results in a NPE if `getScheme` returns null. Switching the order guarantees that `equals` is never called on a null object reference.

File: seedvault/src/main/java/com/solanamobile/seedvault/PublicKeyResponse.java [hunk 2]
```diff
@@ -72,7 +72,12 @@ public PublicKeyResponse(@Nullable byte[] publicKey,
     protected PublicKeyResponse(Parcel in) {
         mPublicKey = in.createByteArray();
         mPublicKeyEncoded = in.readString();
-        resolvedDerivationPath = in.readTypedObject(Uri.CREATOR);
```
Comment (sdlaver) [inline seedvault/src/main/java/com/solanamobile/seedvault/PublicKeyResponse.java:75]: `readTypedObject` can return null, but `resolvedDerivationPath` is marked `@NonNull`

File: seedvault/src/main/java/com/solanamobile/seedvault/SeedVault.java [hunk 3]
```diff
@@ -93,9 +97,11 @@ public static boolean isAvailable(@NonNull Context context, boolean allowSimulat
             return true;
         }
 
-        for (PermissionInfo permission : pi.permissions) {
```
Comment (sdlaver) [inline seedvault/src/main/java/com/solanamobile/seedvault/SeedVault.java:96]: `pi.permissions` can be null

File: seedvault/src/main/java/com/solanamobile/seedvault/SigningRequest.java [hunk 4]
```diff
@@ -44,8 +44,19 @@ public SigningRequest(@NonNull byte[] payload, @NonNull List<Uri> requestedSigna
     }
 
     protected SigningRequest(Parcel in) {
-        payload = in.createByteArray();
```
Comment (sdlaver) [inline seedvault/src/main/java/com/solanamobile/seedvault/SigningRequest.java:47]: `createByteArray` and `createTypedArrayList` can return null, but class fields are marked `@NonNull`

File: seedvault/src/main/java/com/solanamobile/seedvault/SigningResponse.java [hunk 5]
```diff
@@ -49,7 +49,12 @@ protected SigningResponse(Parcel in) {
         for (int i = 0; i < size; i++) {
             mSignatures.add(in.createByteArray());
         }
-        mResolvedDerivationPaths = in.createTypedArrayList(Uri.CREATOR);
```
Comment (sdlaver) [inline seedvault/src/main/java/com/solanamobile/seedvault/SigningResponse.java:52]: `createTypedArrayList` can return null, but `mResolvedDerivationPaths` is marked `@NonNull`

File: seedvault/src/main/java/com/solanamobile/seedvault/Wallet.java [hunk 6]
```diff
@@ -605,7 +605,7 @@ public static boolean hasUnauthorizedSeedsForPurpose(
                 WalletContractV1.UNAUTHORIZED_SEEDS_ALL_COLUMNS,
                 null,
                 null);
-        if (!c.moveToFirst()) {
```
Comment (sdlaver) [inline seedvault/src/main/java/com/solanamobile/seedvault/Wallet.java:608]: `query` can return null

File: seedvault/src/main/java/com/solanamobile/seedvault/Wallet.java [hunk 7]
```diff
@@ -834,7 +834,7 @@ public static ArrayMap<String, Long> getImplementationLimitsForPurpose(
                 WalletContractV1.IMPLEMENTATION_LIMITS_ALL_COLUMNS,
                 null,
                 null);
-        if (!c.moveToNext()) {
```
Comment (sdlaver) [inline seedvault/src/main/java/com/solanamobile/seedvault/Wallet.java:837]: `query` can return null

File: SeedVaultSimulator/src/main/res/drawable/ic_launcher_foreground.xml [hunk 8]
```diff
@@ -1,3 +1,7 @@
+<!--
+  ~ Copyright (c) 2024 Solana Mobile Inc.
```
Comment (J909) [inline SeedVaultSimulator/src/main/res/drawable/ic_launcher_foreground.xml:2]: Do you want to update the (c) 2022 in LICENSE file?

**Review by J909 — APPROVED** [review]

**Pull request merged**